File: minesweeper/python/Cell.py
class Cell:
    def __init__(self, value):
        self._value = value
        self._revealed = False
        self._marked = False

    def display_value(self):
        if (self._marked):
            return '^'

        if not (self._revealed):
            return ' '

        if (self._value == ' '):
            return '.'

        return self._value

    def actual_value(self):
        return self._value

    def reveal(self):
        self._revealed = True

    def is_revealed(self):
        return self._revealed

    def is_marked(self):
        return self._marked

    def is_bomb(self):
        return self._value == '*'

    def has_bombs_around(self, number_of_bombs):
        self._value = number_of_bombs

    def toggle_mine_marking(self):
        if self._marked:
            self._marked = False
        else:
            self._marked = True

File: minesweeper/dart/flutter/minesweeper_flutter_app/lib/src/domain/Cell.dart
class Cell {
  String _value;
  bool _revealed = false;
  bool _marked = false;

  Cell(String value) {
    this._value = value;
  }

  String display_value() {
    if (this._marked) {
      return '^';
    }

    if (!this._revealed) {
      return ' ';
    }

    if (this._value == ' ') {
      return '.';
    }

    return this._value;
  }

  void reveal() {
    this._revealed = true;
  }

  bool is_revealed() {
    return this._revealed;
  }

  bool is_marked() {
    return this._marked;
  }

  bool is_bomb() {
    return this._value == '*';
  }

  void set_display_value(String display_value) {
    this._value = display_value;
  }

  void toggle_mine_marking() {
    if (this._marked) {
      this._marked = false;
    } else {
      this._marked = true;
    }
  }

  String actual_value() {
    return this._value;
  }

  // for debugging
  toString() {
    return this._value;
  }

}
